ci-operator/step-registry/openshift/upgrade/qe/test/canary/openshift-upgrade-qe-test-canary-chain.yaml

```diff
@@ -0,0 +1,24 @@
+chain:
+  as: openshift-upgrade-qe-test-canary
+  steps:
+  - ref: idp-htpasswd
+  - ref: cucushift-pre
+  - ref: openshift-extended-upgrade-pre
+  - ref: cucushift-upgrade-prepare
+  - ref: cucushift-upgrade-prehealthcheck
+  - ref: cucushift-canupgrade-label-node
+  - ref: cucushift-canupgrade-pause-worker
```

Comment:
Per the [discussion](https://redhat-internal.slack.com/archives/GH9JHQFKN/p1721311802315329?thread_ts=1721107446.852209&cid=GH9JHQFKN), we should re-use the existing step - `mco-conf-day2-create-custom-mcp`, we also have similar steps like `cucushift-upgrade-eus-pause-worker-mcp` to pause mcp, we also can modify it to reuse it.

Reply:
`cucushift-upgrade-eus-pause-worker-mcp` paused `worker` pool, but canary upgrade need to pause some of the worker node and then set `maxUnavailable`, so I created a new step.

I don't know `mco-conf-day2-create-custom-mcp` before and I can't access the discussion, I just go through the step, it seems inflexible, we have to setup a json in job config, but my step can work on all cluster and don't care about how many worker nodes it has.

Reply:
> cucushift-upgrade-eus-pause-worker-mcp paused worker pool, but canary upgrade need to pause some of the worker node

That is why I said we need to modify the step to work with the new requirement.

>  and then set maxUnavailable

Per official [doc](https://docs.openshift.com/container-platform/4.16/updating/updating_a_cluster/update-using-custom-machine-config-pools.html), `maxUnavailable` is not a MUST, of course, we can support the option in our developing step, but not with a high priority.

>  I just go through the step, it seems inflexible, we have to setup a json in job config, but my step can work on all cluster and don't care about how many worker nodes it has.

There are a lot of requirements on creating custom mcp. The process would be like, filtering a number of nodes, then add them to a customized name of mcp, and even creating several different mcp including different target nodes. That is why it introduced a  json to configure that. There is a long discussion about how to meet different customers' requirement in https://github.com/openshift/release/pull/54674#discussion_r1688025029 .

A rule of prow ci step is we are trying to create an unify ci, including different sub teams, and even dev teams, rather than users create their own steps with similar functionality.

Reply:
> That is why I said we need to modify the step to work with the new requirement.
If we re-use `cucushift-upgrade-eus-pause-worker-mcp`, I think we'd better update its name as well, because it will not only work for `eus` upgrade, then we have to change `openshift-upgrade-qe-sanity-eus` and `openshift-upgrade-qe-test-eus`, it will affect too many jobs.

Reply:
>  I think we'd better update its name as well

Sure. Actually I discussed with jiajliu to rename the step before, but we did not have real scenario on that time to re-use the step yet (but we foresee it would happen), so we did not rename it yet.

> it will affect too many jobs.

As a steps called by a chain, this is very common. If we can develop high quality of codes to make the function be more robust, I think that is not a problem.